apps/server/src/modules/user/service/user.service.ts

```diff
@@ -224,6 +224,22 @@ export class UserService implements DeletionService, IEventHandler<UserDeletedEv
 		return users;
 	}
 
+	public async findByExternalIdsAndProvidedBySystemId(externalIds: string[], systemId: string): Promise<string[]> {
+		const foundUsers = await this.findByExternalIds(externalIds);
+
+		const verifiedUsers = await this.accountService.findByUserIdsAndSystemId(foundUsers, systemId);
+
+		return verifiedUsers;
+	}
+
+	public async findByExternalIds(externalIds: string[]): Promise<string[]> {
```

Comment:
there already is a findByExternalId further up in this file.
It might make sense to better seperate these two visually, for example naming this one "findMultipleByExternalIds"

Reply:
done